src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            auto const seq = env.seq(alice);
+            auto tx1 = batch::inner(pay(alice, bob, XRP(10)), seq + 1);
+            tx1.removeMember(jss::Account);
+            auto jt = env.jtnofill(
+                batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                tx1,
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2));
+
+            env(jt.jv, batch::sig(bob), ter(telENV_RPC_FAILED));
+            env.close();
+        }
+
+        // Invalid: sfSequence
+        {
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            auto const seq = env.seq(alice);
+            auto tx1 = batch::inner(pay(alice, bob, XRP(10)), seq + 1);
+            tx1.removeMember(jss::Sequence);
+            auto jt = env.jtnofill(
+                batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                tx1,
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2));
+
+            env(jt.jv, batch::sig(bob), ter(telENV_RPC_FAILED));
+            env.close();
+        }
+
+        // Invalid: sfFee
+        {
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            auto const seq = env.seq(alice);
+            auto tx1 = batch::inner(pay(alice, bob, XRP(10)), seq + 1);
+            tx1.removeMember(jss::Fee);
+            auto jt = env.jtnofill(
+                batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                tx1,
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2));
+
+            env(jt.jv, batch::sig(bob), ter(telENV_RPC_FAILED));
+            env.close();
+        }
+
+        // Invalid: sfSigningPubKey
+        {
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            auto const seq = env.seq(alice);
+            auto tx1 = batch::inner(pay(alice, bob, XRP(10)), seq + 1);
+            tx1.removeMember(jss::SigningPubKey);
+            auto jt = env.jtnofill(
+                batch::outer(alice, seq, batchFee, tfAllOrNothing),
+                tx1,
+                batch::inner(pay(alice, bob, XRP(10)), seq + 2));
+
+            env(jt.jv, batch::sig(bob), ter(telENV_RPC_FAILED));
+            env.close();
+        }
+    }
+
+    void
+    testBadSequence(FeatureBitset features)
+    {
+        testcase("bad sequence");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, gw);
+        env.close();
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        env(noop(bob), ter(tesSUCCESS));
+        env.close();
+
+        // Invalid: Bob Sequence is a past sequence
```

Comment:
Can you add tests that are identical to the first two in this section, but with Alice instead of Bob (Alice still submitting the outer transaction)?